hedera-mirror-rest/controllers/contractController.js

```
@@ -526,6 +555,47 @@ const extractContractResultsByIdQuery = (filters, contractId, paramSupportMap =
           contractResultTimestampFullName
         );
         break;
+      case constants.filterKeys.BLOCK_NUMBER:
+      case constants.filterKeys.BLOCK_HASH:
```

If both `block.number` and `block.hash` are provided it should throw an error.